src/api/fetchData.js

```diff
@@ -2,24 +2,29 @@ import { useState, useEffect } from 'react';
 
 export default function fetchData(url) {
     const [data, setData] = useState([]);
-    const [loading, setLoading] = useState(null);
+    const [loading, setLoading] = useState(true);
```

Comment:
I think loader starting in true is a very good change. Then when the request starts, it can be better to be set to true again and finally to false again.

This is because after a first request when it's finished, the state of loading will be set to false in the finally statement. And in a second request it will start as false because of this, and it should start as true.

I think the setLoading(true) in line 9 inside useEffect had this effect, leaving it would solve this.

Reply:
I decided not to add the additional setLoading(true) because it leaves the app in an infinite loop of fetching the data and refreshing the component.
It's true that we shoul deal with concurrent calls, but I was hoping we could find a way to fix this in the future.